src/core/radio/radio.hpp

```diff
@@ -67,6 +69,58 @@ static constexpr uint64_t kMaxCslTimeout = OPENTHREAD_CONFIG_MAC_CSL_MAX_TIMEOUT
  *
  */
 
+/**
+ * This class implements the radio statistics logic.
+ *
+ * The radio statistics are the time when the radio in TX/RX/radio state.
+ * Since this class collects these statistics from pure software level and no platform API is involved, a simplied
+ * model is used to calculate the time of different radio states. The data may not be very accurate, but it's
+ * sufficient to provide a general understanding of the proportion of time a device is in different radio states.
+ *
+ * The simplified model is:
+ * - The RadioStats is only aware of 2 states: RX and sleep.
+ * - Each time `Radio::Receive` or `Radio::Sleep` is called, it will check the current state and add the time since
+ *   last time the methods were called. For example, `Sleep` is first called and `Receive` is called after 1 second,
+ *   then 1 second will be added to SleepTime and the current state switches to `Receive`.
+ * - The time of TX will be calculated from the callback of TransmitDone. If TX returns OT_ERROR_NONE or
+ *   OT_ERROR_NO_ACK, the tx time will be added according to the number of bytes sent. And the SleepTime or RxTime
+ *   will be reduced accordingly.
+ * - When `GetStats` is called, an operation will be executed to calcute the time for the last state. And the result
+ *   will be returned.
+ *
+ */
+#if OPENTHREAD_CONFIG_RADIO_TIME_STATS_ENABLE && (OPENTHREAD_FTD || OPENTHREAD_MTD)
+class RadioStatistics
+{
+public:
+    enum Status : uint8_t
+    {
+        kDisabled,
+        kSleep,
+        kReceive,
+    };
+
+    explicit RadioStatistics(void);
+
+    void ChangeStatus(Status aStatus);
+    void HandleReceiveAt(uint32_t aDurationUs);
+    void TxDone(otError aError, uint16_t aPsduLength);
+    void RxDone(otError aError);
+    void GetStats(otRadioStats &aRadioStats);
```

Comment:
If we define and use a `mStats` memeber variable we can change this API to return a `const` reference to it directly?
```
const otRadioStats &GetStats(void) const;
```